### PR TITLE
augustus: Set compile commands for each compiler and Fix for using 'boost' on Spack

### PR DESCRIPTION
**Pull request by t-karatsu**
* `augustus` always uses only gnu compiler, so I fixed makefiles to use compile commands of specified compiler on Spack.
* Fix to use `boost` installed by Spack.

### CODE REVIEW

File: var/spack/repos/builtin/packages/augustus/package.py [hunk 1]
```diff
@@ -84,6 +84,34 @@ def edit(self, spec, prefix):
...
+            'auxprogs/compileSpliceCands/Makefile',
+            'auxprogs/homGeneMapping/src/Makefile',
+            'auxprogs/joingenes/Makefile',
+            'src/Makefile'
+        ]
+        for makefile in makefiles:
+            filter_file('gcc', '{0}'
+                        .format(spack_cc), makefile, string=True)
+            filter_file('g++', '{0}'
+                        .format(spack_cxx), makefile, string=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/augustus/package.py]: No need to use string formatting here, `spack_cc` and friends are already strings.

Comment (t-karatsu) [reply var/spack/repos/builtin/packages/augustus/package.py]: Thanks for review. `g++` is included regex, so I deleted `string` arg from only `filter_file` func of 110 line.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/augustus/package.py]: Actually, `string=True` is fine, what I meant was you can do:
```python
filter_file('gcc', spack_cc, makefile, string=True)
```
There's no need to convert `spack_cc` to a string.

Comment (t-karatsu) [reply var/spack/repos/builtin/packages/augustus/package.py]: Sorry... I was misunderstanding. I fixed it.

File: var/spack/repos/builtin/packages/augustus/package.py [hunk 2]
```diff
@@ -84,6 +84,32 @@ def edit(self, spec, prefix):
             makefile.filter('$(HTSLIB)/libhts.a',
                             '$(HTSLIB)/../lib/libhts.a', string=True)
 
+        # Set compile commands for each compiler and
+        # Fix for using 'boost' on Spack. (only after ver.3.3.1-tag1)
+        if self.version >= Version('3.3.1-tag1'):
+            with working_dir(join_path('auxprogs', 'utrrnaseq', 'Debug')):
+                filter_file('g++',
+                            '{0}'.format(spack_cxx), 'makefile', string=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/augustus/package.py]: Same here

Comment (t-karatsu) [reply var/spack/repos/builtin/packages/augustus/package.py]: It's my oversight... Fixed it.

**Pull request merged**